### ISSUE TITLE
Error thrown when opening a ComboboxControl within InspectorControls

### ISSUE THREAD

**Issue by jtrost** [WordPress/gutenberg#29566]
## Description
Opening a `<ComboboxControl>` within `<InspectorControls>` throws an error. This does not happen within the editor itself.
````
Uncaught TypeError: u is not a function
onFocus http://localhost:8888/wordpress/wp-includes/js/dist/components.min.js?ver=acda7ed704a5cd88b3f144fad5e2edec:7
````

The error points to this minified function:
````javascript
onFocus:function(){C(!0),u(""),T("")}
````

It appears that this error is saying that `onFilterValueChange` is not a function [here](https://github.com/WordPress/gutenberg/blob/a68d4a2c5084af930590c0eb000453a04924bb97/packages/components/src/combobox-control/index.js#L139).

Functionally, nothing is broken. I can still select an option from the Combobox and save it. But I don't want to distribute a block that throws an error.

## Step-by-step reproduction instructions
Example Code:
````html
<InspectorControls>
  <PanelBody title="Button Attributes" initialOpen={true}>
    <PanelRow>
      <ComboboxControl
        label="Action"
        value={attributes.action}
        onChange={(val) => setAttributes({action: val})}
        options={[{value: "submit", label: "Submit"}, {value: "previous", label: "Previous Step"},  {value: "next", label: "Next Step"}]}
      />
    </PanelRow>
  </PanelBody>
</InspectorControls>
````

## Expected behaviour
No error is thrown.

## Actual behaviour
An error is thrown.

## WordPress information
- WordPress version: 5.6.1
- Gutenberg version: Not listed in /wp-admin/site-health.php?tab=debug
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

## Device information
- Device: Desktop
- Operating system: OS X
- Browser: Firefox 86.0

**Comment by talldan** [comment]
@jtrost Looks like this was fixed in https://github.com/WordPress/gutenberg/pull/28492. `onFilterValueChange` was originally mistakenly implemented as a required prop, and it's now optional.

You should be able to fix this in your code by passing a noop function (`() => {}`) for the `onFilterValueChange` prop.

**Comment by JohnRDOrazio** [comment]
The examples given show the usage of `onInputChange` alongside `onChange`, I believe this is an error?

https://github.com/WordPress/gutenberg/blame/trunk/packages/components/src/combobox-control/README.md#L52-L54

The instantiation of the Component allows for `onChange` and `onFilterValueChange`:
https://github.com/WordPress/gutenberg/blob/7c7f8663fb6f9cbcd6a070095308f0de52c3e855/packages/components/src/combobox-control/index.js#L48-L49

If I define `onInputChange` as in the example, I get the console error `Uncaught TypeError: onFilterValueChange is not a function`. If I define `onFilterValueChange` instead of `onInputChange` I no longer get the error.

And I see that the `onChange` event on the component is calling `onInputChange`:
https://github.com/WordPress/gutenberg/blob/7c7f8663fb6f9cbcd6a070095308f0de52c3e855/packages/components/src/combobox-control/index.js#L218
 which in turn is calling `onFilterValueChange`:
https://github.com/WordPress/gutenberg/blob/7c7f8663fb6f9cbcd6a070095308f0de52c3e855/packages/components/src/combobox-control/index.js#L150.
`onFilterValueChange` however doesn't have an internal definition but depends on whatever function is passed to it, whereas `onInputChange` has it's own internal definition so I believe shouldn't be overridden?

Basically I'm pretty sure the example needs to be updated from `onInputChange` to `onFilterValueChange` in the component parameters.
